src/tests.rs

```diff
@@ -4217,6 +4218,198 @@ fn test_key() {
     );
     assert_eq!(ret, CKR_OK);
 
+    /* Test key derivation */
```

Comment:
Can we have it in separate function? This one is terribly long already:
 * when it fails one needs to dig deeper into the code whats the issue
 * these tests cant be executed in parallel if it is infinite spaghetti code

Reply:
I have another patch coming that adds a whole bunch of things.
I am thinking that I want to add more spaghetti in this PR, and then work on a separate PR to completely reorganize the test code in separate files.
Do you mind bearing with me until then :)